File: server/plugin/pcdn/router/alarm.go
package router

import "github.com/gin-gonic/gin"

// AlarmRouter 告警路由（admin 组）
type AlarmRouter struct{}

// InitAlarmRouter 挂载告警规则与记录路由
func (r *AlarmRouter) InitAlarmRouter(group *gin.RouterGroup) {
	group.GET("alarm/rule/list", alarmApi.GetAlarmRuleList)
	group.GET("alarm/rule/find", alarmApi.GetAlarmRule)
	group.POST("alarm/rule/create", alarmApi.CreateAlarmRule)
	group.PUT("alarm/rule/update", alarmApi.UpdateAlarmRule)
	group.DELETE("alarm/rule/delete", alarmApi.DeleteAlarmRule)
	group.DELETE("alarm/rule/deleteByIds", alarmApi.DeleteAlarmRuleByIds)
	group.GET("alarm/record/list", alarmApi.GetAlarmRecordList)
}

File: web/src/plugin/pcdn/api/alarm.js
import service from '@/utils/request'

// 告警规则
export const getAlarmRuleList = (params) => service({ url: '/pcdn/admin/alarm/rule/list', method: 'get', params })
export const getAlarmRule = (params) => service({ url: '/pcdn/admin/alarm/rule/find', method: 'get', params })
export const createAlarmRule = (data) => service({ url: '/pcdn/admin/alarm/rule/create', method: 'post', data })
export const updateAlarmRule = (data) => service({ url: '/pcdn/admin/alarm/rule/update', method: 'put', data })
export const deleteAlarmRule = (params) => service({ url: '/pcdn/admin/alarm/rule/delete', method: 'delete', params })
export const deleteAlarmRuleByIds = (data) => service({ url: '/pcdn/admin/alarm/rule/deleteByIds', method: 'delete', data })

// 告警记录
export const getAlarmRecordList = (params) => service({ url: '/pcdn/admin/alarm/record/list', method: 'get', params })
